### ISSUE TITLE
DAG Scoring: `get_labels`'s mapping is incorrect

### ISSUE THREAD

**Issue by Caceresenzo** [crunchdao/crunch-cli#103]
wildebeest_42 (on discord) reported an issue with invalid mapping when processing DAG predictions.

The problem is in the bit-shift optimization, which is done in the wrong order:
https://github.com/crunchdao/crunch-cli/blob/bca1575d6c21055d14190504726aaf489f3f9104/crunch/scoring/_format/dag.py#L82-L87

Instead, it should be:
```python
mapping[
    (A[x_index, index] << 3) |
    (A[y_index, index] << 2) |
    (A[index, y_index] << 1) |
    (A[index, x_index])
]
```

Scoring is not affected because even if the mapping was wrong, the classification was still unique:
![image](https://github.com/user-attachments/assets/36dbc330-3d08-4d6b-9173-3422b9910de8)

**Comment by jeheran** [comment]
Can we run 10 unique submission?

**Comment by Caceresenzo** [comment]
The only differences are floating precision errors.

| Prediction ID | `get_labels` | Score |
| --- | --- | --- |
| 14145 | original | **0.3789**7837728714573 |
| | crunchcli | **0.3789**7837728714573 |
| | fixed | **0.3789**783772871458 |
| | | |
| 14397 | original | **0.1926**8937435035088 |
| | crunchcli | **0.1926**8937435035088 |
| | fixed | **0.1926**8937435035088 |
| | | |
| 14398 | original | **0.3379**422612210875 |
| | crunchcli | **0.3379**422612210875 |
| | fixed | **0.3379**422612210875 |
| | | |
| 14400 | original | **0.4914**5823973035085 |
| | crunchcli | **0.4914**5823973035085 |
| | fixed | **0.4914**5823973035085 |
| | | |
| 14439 | original | **0.5087**069557397074 |
| | crunchcli | **0.5087**069557397074 |
| | fixed | **0.5087**069557397074 |
| | | |
| 14502 | original | **0.4508**791902673268 |
| | crunchcli | **0.4508**791902673268 |
| | fixed | **0.4508**7919026732687 |
| | | |
| 14721 | original | **0.5086**765662451519 |
| | crunchcli | **0.5086**765662451519 |
| | fixed | **0.5086**765662451519 |
| | | |
| 14792 | original | **0.5247**932503628641 |
| | crunchcli | **0.5247**932503628641 |
| | fixed | **0.5247**932503628641 |
| | | |
| 14798 | original | **0.5352**476743966134 |
| | crunchcli | **0.5352**476743966134 |
| | fixed | **0.5352**476743966134 |
| | | |
| 15011 | original | **0.5212**374326452135 |
| | crunchcli | **0.5212**374326452135 |
| | fixed | **0.5212**374326452135 |
| | | |
| 15038 | original | **0.3786**8331031662406 |
| | crunchcli | **0.3786**8331031662406 |
| | fixed | **0.3786**833103166241 |
| | | |
| 15326 | original | **0.3790**470402968087 |
| | crunchcli | **0.3790**470402968087 |
| | fixed | **0.3790**4704029680863 |
